example/pokeapi_functional/lib/controllers/pokemon_provider.dart

```diff
@@ -1,25 +1,16 @@
 import 'package:fpdart/fpdart.dart';
-import 'package:hooks_riverpod/hooks_riverpod.dart';
-import 'package:pokeapi_functional/api/fetch_pokemon.dart';
-import 'package:pokeapi_functional/constants/constants.dart';
-import 'package:pokeapi_functional/models/pokemon.dart';
-import 'package:pokeapi_functional/unions/request_status.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
 
-/// Manage the [Pokemon] state using [Either] ([TaskEither]) to handle possible errors.
-///
-/// Each [RequestStatus] changes the UI displayed to the user.
-class PokemonState extends StateNotifier<RequestStatus> {
-  PokemonState() : super(const RequestStatus.initial());
+import '../api/fetch_pokemon.dart';
+import '../models/pokemon.dart';
 
-  /// Initial request, fetch random pokemon passing the pokemon id.
-  Future<Unit> fetchRandom() async => _pokemonRequest(
-        () => fetchPokemon(
-          randomInt(
-            Constants.minimumPokemonId,
-            Constants.maximumPokemonId + 1,
-          ).run(),
-        ),
-      );
+part 'pokemon_provider.g.dart';
+
+@riverpod
+class PokemonState extends _$PokemonState {
+  @override
+  FutureOr<Pokemon> build() async => (await fetchRandomPokemon().run())
+      .match((error) => throw error, (pokemon) => pokemon);
```

Comment:
Instead of `run()` then `match()`, `fetchRandomPokemon` returns a `TaskEither` on which you could call `match` directly.

However, in this case you can use `getOrElse` instead of `match`.

Reply:
What happens if the `build` method throws?

Reply:
Return [AsyncError](https://pub.dev/documentation/riverpod/latest/riverpod/AsyncError-class.html) if thrown by the build method

https://pub.dev/documentation/riverpod/latest/riverpod/AsyncNotifier/build.html